https://medium.com/interfaith-now/does-evidence-for-jesus-resurrection-meet-hume-s-criteria-for-miracles-b23b069dd2e7

In [1]:
from bayes_net import *

In [2]:
T=Variable("Theism",[True,False])
T.set_cpt([0.5,0.5])
T.cpt

  Theism       p      
  ======       =      
   True     0.500000  
  False     0.500000  

In [3]:
M=Variable("Messiah",[True,False],parents=[T])
M.set_cpt([0.25,0.75,0,1])
M.cpt

  Theism    Messiah       p      
  ======    =======       =      
   True       True     0.250000  
   True      False     0.750000  
  False       True     0.000000  
  False      False     1.000000  

> Therefore, the prior probability of a Messiah p(M|K) is 12.5%

This is sloppy.  He should have P(T|K)=0.5, P(M|K,T)=0.25, etc...

In [4]:
net=Network([T,M])

In [5]:
net.P("Messiah")

 Messiah       p      
 =======       =      
   True     0.125000  
  False     0.875000  

Need to marginalize over the $T$, like:

\begin{aligned}
P(M)&=P(M|T)P(T) + P(M|\sim\! T)P(\sim\! T) \\
&=0.25 \cdot 0.5 + 0 \cdot 0.5 \\
&=0.125
\end{aligned}

> The prior probability of a resurrection given our background knowledge p(R|K) is 0.03 (3%):
>
> - p(R|M&K) — probability of a resurrection if there is a Messiah multiplied (0.25) by
> - p(M|T) — probability of a Messiah if there is a God (0.25) multiplied by
> - p(T) — probability of God’s existence (0.5)

again, sloppy

In [6]:
R=Variable("Resurrection",[True,False],parents=[M])
R.set_cpt([0.25,0.75,0,1])
R.cpt

 Messiah   Resurrection     p      
 =======   ============     =      
   True       True     0.250000  
   True      False     0.750000  
  False       True     0.000000  
  False      False     1.000000  

In [7]:
net=Network([T,M,R])

In [8]:
net.P("Resurrection")

Resurrection     p      
============     =      
   True     0.031250  
  False     0.968750  

In [9]:
EM=Variable("EvidenceM",[True,False],parents=[M])
EM.set_cpt([0.1,0.9,0.001,.999])
EM.cpt

 Messiah   EvidenceM      p      
 =======   =========      =      
   True       True     0.100000  
   True      False     0.900000  
  False       True     0.001000  
  False      False     0.999000  

In [10]:
net=Network([T,M,R,EM])

In [11]:
net.P("EvidenceM and Messiah")

 Messiah   EvidenceM      p      
 =======   =========      =      
   True       True     0.012500  
   True      False     0.112500  
  False       True     0.000875  
  False      False     0.874125  

In [12]:
ER=Variable("EvidenceR",[True,False],parents=[R])
ER.set_cpt([0.1,0.9,0.001,.999])
ER.cpt

Resurrection EvidenceR      p      
============ =========      =      
   True       True     0.100000  
   True      False     0.900000  
  False       True     0.001000  
  False      False     0.999000  

In [13]:
net=Network([T,M,R,EM,ER])

In [14]:
net

P(Theism,Messiah,Resurrection,EvidenceM,EvidenceR)=P(Theism)P(Messiah|Theism)P(Resurrection|Messiah)P(EvidenceM|Messiah)P(EvidenceR|Resurrection)

P(Theism):

  Theism       p      
  ======       =      
   True     0.500000  
  False     0.500000  

P(Messiah|Theism):

  Theism    Messiah       p      
  ======    =======       =      
   True       True     0.250000  
   True      False     0.750000  
  False       True     0.000000  
  False      False     1.000000  

P(Resurrection|Messiah):

 Messiah   Resurrection     p      
 =======   ============     =      
   True       True     0.250000  
   True      False     0.750000  
  False       True     0.000000  
  False      False     1.000000  

P(EvidenceM|Messiah):

 Messiah   EvidenceM      p      
 =======   =========      =      
   True       True     0.100000  
   True      False     0.900000  
  False       True     0.001000  
  False      False     0.999000  

P(EvidenceR|Resurrection):

Resurrection EvidenceR      p   

In [15]:
net.P("Resurrection")

Resurrection     p      
============     =      
   True     0.031250  
  False     0.968750  

In [19]:
net.P("Resurrection|EvidenceR and EvidenceM")

Resurrection     p      
============     =      
   True     0.968242  
  False     0.031758  

In [21]:
net.P("Theism|EvidenceR and EvidenceM")

  Theism       p      
  ======       =      
   True     0.998451  
  False     0.001549  

In [22]:
net.P("Messiah|EvidenceR and EvidenceM")

 Messiah       p      
 =======       =      
   True     0.997289  
  False     0.002711  